### PR TITLE
Authentication docs improvements

### PR DESCRIPTION
**Pull request by CTNicholas**
- https://github.com/liveblocks/liveblocks.io/pull/1991

### CODE REVIEW

File: docs/pages/authentication.mdx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+---
+
+Liveblocks provides different methods to authenticate your application using
+your public and secret API keys. For any production application, you should use
+your secret key to enable access token or ID token authentication. These methods
+can be used to control access to your rooms and data.
+
+<Banner title="Public key is for prototyping" type="warning">
+
+We don’t recommend using your public API key in production, as it’s difficult
```
Comment (ofoucherot) [inline docs/pages/authentication.mdx:14]: I would also say that it's not secure.

File: docs/pages/authentication.mdx [hunk 2]
```diff
@@ -0,0 +1,108 @@
+---
```
Comment (ofoucherot) [inline docs/pages/authentication.mdx:1]: Love this page!

I think we should add a section: `how to choose between ID vs Access tokens`.

Access tokens method requires less work to set up, but it has some limitations (the rooms need to be organized in a tree).

File: docs/pages/authentication/access-token/nextjs.mdx [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+        // Start an auth session inside your endpoint
+        const session = liveblocks.prepareSession(
+          user.id,
+          { userInfo: user.metadata } // Optional
+        );
+
+        // Implement your own security, and give the user access to the room/organization.
+        // Note: Even if room is defined, we recommend to always use wildcards.
+        const { room } = request.body;
```
Comment (ofoucherot) [inline docs/pages/authentication/access-token/nextjs.mdx:61]: Should we use wildcards as the main example (even if room is set)?

And add another section to show how to generate token per room/

File: docs/pages/authentication/id-token.mdx [hunk 4]
```diff
@@ -0,0 +1,378 @@
+---
+meta:
+  title: "Authenticate with ID tokens"
+  parentTitle: "Authentication"
+  description: "Learn more about ID token permissions"
+---
+
+ID token authentication allows Liveblocks to handle permissions for you. This
+means that when you create or modify a room, you can set a user’s permissions on
+the room itself, this acting as a source of truth. Later, when a user tries to
```
Comment (FlowFlorent) [inline docs/pages/authentication/id-token.mdx:10]: "Groups" could be mentioned here as well, not necessarily explained but at least they'd know what to expect when clicking on the ID token page.

File: docs/pages/authentication.mdx (hunk 2 above)
Comment (CTNicholas) [reply docs/pages/authentication.mdx:1]: Great idea!

File: docs/pages/authentication/id-token.mdx (hunk 4 above)
Comment (CTNicholas) [reply docs/pages/authentication/id-token.mdx]: Yes! Done it.

File: docs/pages/authentication/access-token/nextjs.mdx [hunk 5]
```diff
@@ -56,22 +88,22 @@ start building your own security logic in Next.js’ `/app` directory.
           { userInfo: user.metadata } // Optional
         );
 
-        // Implement your own security, and give the user access to the room/organization.
-        // Note: Even if room is defined, we recommend to always use wildcards.
         const { room } = request.body;
-        if (room && __shouldUserHaveAccess__(user, room)) {
+
+        if (room && **shouldUserHaveAccess**(user, room)) {
```
Comment (marcbouchenoire) [inline docs/pages/authentication/access-token/nextjs.mdx:93]: Maybe some Markdown auto-formatting result? This should stay `__`.

**Review by ofoucherot — APPROVED** [review]

**Pull request merged**